src/FunctionThrottler.js

```diff
@@ -0,0 +1,15 @@
+export class FunctionThrottler {
```

Comment:
didn't https://www.npmjs.com/package/just-throttle work?

Reply:
:+1:

Reply:
I am not pretty sure about if is this going to work in the same way:
Their code does the follow:
```javascript
function throttle(fn, interval, callFirst) {
  var wait = false;
  var callNow = false;
  return function() {
    callNow = callFirst && !wait;
    var context = this;
    var args = arguments;
    if (!wait) {
      wait = true;
      setTimeout(function() {
        wait = false;
        if (!callFirst) {
          return fn.apply(context, args);
        }
      }, interval);
    }
    if (callNow) {
      callNow = false;
      return fn.apply(this, arguments);
    }
  };
```

If I am okay what this is going to do is:
- If is first time execution -> call fn
- If is second time execution -> delay it XXX ms and block any other execution

What our FN does is:
-  First time execution -> call fn
-  If is second time execution and there are no subsequent calls -> call fn
-  If is third time execution after and haven't elapsed our config time don't do nothing.

Their function and ours is not doing the same. Our function calls always at time but block next calls, their function block the call and subsequent calls for XX ms and then mades the fn call.

I don't think that we'll have any problem to use just-throttle but I don't know If we could reach some side effect for this change.

Reply:
A quick look of our code, it seems that is doing the first call right away and the second one is ignored is hasn't passed time enough. 

I think we should use debounce function instead: https://www.npmjs.com/package/just-debounce with the flag to `true`.

Reply:
Yes I misspelled the second point in any case I'll delete the throttle and add `just-debounce` as a dependency.